packages/editor/src/components/post-locked-modal/index.js

```diff
@@ -30,17 +30,30 @@ class PostLockedModal extends Component {
 	}
 
 	componentDidMount() {
+		const hookNamePrefix = this.getHookNamePrefix();
+
 		// Details on these events on the Heartbeat API docs
 		// https://developer.wordpress.org/plugins/javascript/heartbeat-api/
-		jQuery( document )
-			.on( 'heartbeat-send.refresh-lock', this.sendPostLock )
-			.on( 'heartbeat-tick.refresh-lock', this.receivePostLock );
+		addAction( 'heartbeat.send', hookNamePrefix + '-send', this.sendPostLock );
+		addAction( 'heartbeat.tick', hookNamePrefix + '-tick', this.receivePostLock );
 	}
 
 	componentWillUnmount() {
-		jQuery( document )
-			.off( 'heartbeat-send.refresh-lock', this.sendPostLock )
-			.off( 'heartbeat-tick.refresh-lock', this.receivePostLock );
+		const hookNamePrefix = this.getHookNamePrefix();
+
+		removeAction( 'heartbeat.send', hookNamePrefix + '-send' );
+		removeAction( 'heartbeat.tick', hookNamePrefix + '-tick' );
+	}
+
+	/**
+	 * Returns a `@wordpress/hooks` hook name prefix specific to the instance
+	 * of the component.
+	 *
+	 * @return {string} Hook name prefix.
+	 */
+	getHookNamePrefix() {
+		const { instanceId } = this.props;
+		return 'core/editor/post-locked-modal-' + instanceId + '-heartbeat';
```

Comment:
I like this namespace better. Yet I find the `heartbeat-{send/tick}` repetition not necessary as it is already in the action name. How about:

```suggestion
		return 'core/editor/post-locked-modal-' + instanceId;
```

Reply:
FWIW, that would be more in line with the other existing use case (nonce middleware) where we use `core/api-fetch/create-nonce-middleware`.

Reply:
Sure 👍 Since the function name needs to be provided when calling `removeAction` anyways (the theoretical need for this in the first place), it's arguably redundant.